cmd/config.go

```diff
@@ -309,17 +310,30 @@ func getConsolidatedConfig(fs afero.Fs, cliConf Config, runner lib.Runner) (conf
 	conf = conf.Apply(envConf).Apply(cliConf)
 	conf = applyDefault(conf)
 
+	// TODO(imiric): Move this validation where it makes sense in the configuration
+	// refactor of #883. Currently we validate CLI flags in cmd.getOptions. Yet
+	// there is no place where configuration is validated after applying all other
+	// sources, like environment variables (besides inline in the runCmd itself...).
+	// So this repeats the trend stats validation in case other sources overrode our
+	// default value.
+	if err = ui.ValidateSummary(conf.SummaryTrendStats); err != nil {
+		return conf, err
+	}
+
 	return conf, nil
 }
 
-// applyDefault applys default options value if it is not specified by any mechenisms. This happens with types
-// which does not support by "gopkg.in/guregu/null.v3".
+// applyDefault applies the default options value if it is not specified.
+// This happens with types which are not supported by "gopkg.in/guregu/null.v3".
 //
 // Note that if you add option default value here, also add it in command line argument help text.
 func applyDefault(conf Config) Config {
 	if conf.Options.SystemTags == nil {
 		conf = conf.Apply(Config{Options: lib.Options{SystemTags: &stats.DefaultSystemTagSet}})
 	}
+	if conf.Options.SummaryTrendStats == nil {
+		conf = conf.Apply(Config{Options: lib.Options{SummaryTrendStats: lib.DefaultSummaryTrendStats}})
+	}
```

Comment:
Hmm I didn't catch this the first time, but these `conf.Apply()` seem unnecessary :confused: Can't we just have:
```go
	if conf.Options.SystemTags == nil {
		conf.Options.SystemTags = &stats.DefaultSystemTagSet
	}
	if conf.Options.SummaryTrendStats == nil {
		conf.Options.SummaryTrendStats =  lib.DefaultSummaryTrendStats
	}
```

Reply:
Indeed! Fixed in b1b488ec.